rest/src/main/java/com/confighub/api/repository/client/AClientAccessValidation.java

```diff
@@ -209,34 +195,41 @@ public void validatePull(String clientToken,
         {
             throw new ConfigException(Error.Code.PARTIAL_CONTEXT);
         }
-
-        addJsonHeader(json, repository, contextString, context);
-        long start = System.currentTimeMillis();
-        resolved = resolveForClient();
-
-        log.info("Client [%s] from [%s] resolved %d keys in %d/ms > %s",
-                 appName,
-                 remoteIp,
-                 resolved.size(),
-                 (System.currentTimeMillis() - start),
-                 contextString);
-
-
-        processAuth(store, gson, securityProfiles, token, repository, date, passwords);
     }
 
-    private ImmutableMap<PropertyKey, Property> resolveForClient()
+    public void validatePull(String clientToken,
+                             String contextString,
+                             String version,
+                             String appName,
+                             String remoteIp,
+                             Store store,
+                             Gson gson,
+                             String securityProfiles,
+                             boolean validateAuthOnly)
+            throws ConfigException
     {
-        if (!repository.isCachingEnabled())
-        {
-            return ImmutableMap.copyOf(context.resolveForClient());
-        }
+        checkToken(clientToken, store);
```

Comment:
by splitting out the resolveContext we are not checking the Token after we do some work processing the context.  Wouldn't we want to vet out the token before processing anything.  Also questions whether we should vet out auth before processing anything (seems like that was already there though).  Ideally if this was a public facing website we'd want to prevent anyone without access from hammering the server with things to do.